archaius-guice/src/main/java/com/netflix/archaius/guice/ArchaiusModule.java

```diff
@@ -139,37 +138,39 @@ public void afterInjection(I injectee) {
             return injector.getInstance(Key.get(type, Names.named(name)));
         }
     }
+
+    private final AppConfig appConfig;
+    
+    public ArchaiusModule() {
+        this(null);
+    }
+    
+    public ArchaiusModule(AppConfig appConfig) {
+        this.appConfig = appConfig;
+    }
+    
+    public static ArchaiusModule fromAppConfig(final AppConfig config) {
```

Comment:
What is the rationale having the public constructor that takes an `AppConfig` and this method?